shaas.go

```diff
@@ -333,13 +338,32 @@ func (fww flushWriterWrapper) Write(p []byte) (n int, err error) {
 	return
 }
 
-func httpPort() string {
+func primaryHTTPPort() string {
 	if port := os.Getenv("PORT"); port != "" {
 		return port
 	}
 	return "5000"
 }
 
+func httpPorts() []string {
+	ports := map[string]bool{
```

Comment:
why create the `map[string]bool` first? any reason you can't just use the `[]string` straight away? not seeing what the boolean is doing for us...

Reply:
Ah, it's to cover the case when PORT and ADDITIONAL_HTTP_PORTS overlap. Say ADDITIONAL_HTTP_PORTS was 7000,8000 and PORT also got dynamically assigned to 8000. Added a comment to the function.

Reply:
ah, i see, it's not about the truthiness, but just being used as a unique set. perhaps make it an `map[string]struct{}` so that's clearer? otherwise, looks good.

Reply:
Done!